### ISSUE TITLE
Three different rules trigger on perfectly legal function definition

### ISSUE THREAD

**Issue by mehes-kth** [koalaman/shellcheck#2347]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2275, SC1036, SC1088
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

==() {
        echo "I'm a legal function"
}
```

#### Here's what shellcheck currently says:

```console
$ shellcheck myscript
 
Line 3:
==() {
^-- SC2275: Command name starts with =. Bad line break?
  ^-- SC1036: '(' is invalid here. Did you forget to escape it?
  ^-- SC1088: Parsing stopped here. Invalid use of parentheses?

$ 
```

#### Here's what I wanted or expected to see:

Code passes without factually wrong warnings (meaning the first two; the third one is correct, although it shouldn't be).

**Comment by mehes-kth** [comment]
Interestingly, prefixing the function definition with the _optional_ directive `function` makes this pass, indicating that there's nothing actually wrong with the sample...

**Comment by mehes-kth** [comment]
That really helps, thanks!!!
